pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,661 @@
+					},
+				},
+				Status: tenancyv1beta1.WorkspaceStatus{
+					URL: homeWorkspaceURL.String(),
+				},
+			}
+		}
+
+		responsewriters.WriteObjectNegotiated(homeWorkspaceCodecs, negotiation.DefaultEndpointRestrictions, tenancyv1beta1.SchemeGroupVersion, rw, req, http.StatusOK, homeWorkspace)
+		return
+	}
+
+	needsAutomaticCreation, workspaceType := h.needsAutomaticCreation(workspaceName)
+	if needsAutomaticCreation {
+		isHome := workspaceType == "Home"
+		var retryAfterSeconds int
+		var retryLater bool
+		var localCheckError error
+		var foundLocally bool
+		foundLocally, retryLater, retryAfterSeconds, localCheckError = h.checkWorkspaceLocally(workspaceName, isHome, user.GetName())
+		if localCheckError != nil {
+			responsewriters.InternalError(rw, req, localCheckError)
+			return
+		}
+
+		if !foundLocally {
+			// Home or bucket workspace not found in the local informer
+			// Let's try to create it
+
+			// But first check we have the right to do so.
+			attributes := homeWorkspaceAuthorizerAttributes(user, "create")
+
+			if isHome && workspaceName != h.getHomeWorkspaceName(user.GetName()) {
+				// If we're checking a home workspace or home bucket workspace, but not of the consistent user, let's refuse.
+				responsewriters.Forbidden(ctx, attributes, rw, req, "home workspace can only be created by the user of the home workspace", homeWorkspaceCodecs)
+				return
+			}
+
+			// Test if the user has the right to create his Home workspace when it doesn't exists
+			// => test the create verb on the clusterworkspaces/workspace subresource named ~ in the root workspace.
+
+			if decision, reason, err := h.authz.Authorize(
+				request.WithCluster(ctx, request.Cluster{Name: tenancyv1alpha1.RootCluster}),
+				attributes); err != nil {
+				responsewriters.InternalError(rw, req, err)
+				return
+			} else if decision != authorizer.DecisionAllow {
+				responsewriters.Forbidden(ctx, attributes, rw, req, reason, homeWorkspaceCodecs)
+				return
+			}
+
+			var createError error
+			retryLater, retryAfterSeconds, createError = h.tryToCreate(ctx, user.GetName(), workspaceName, workspaceType)
+			if createError != nil {
+				responsewriters.ErrorNegotiated(createError, errorCodecs, schema.GroupVersion{}, rw, req)
+				return
+			}
+		}
+
+		if retryLater {
+			// Return a 429 status asking the client to try again after the creationDelay
+			rw.Header().Set("Retry-After", fmt.Sprintf("%d", retryAfterSeconds))
+			http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
+			return
+		}
+	}
+	h.apiHandler.ServeHTTP(rw, req)
+}
+
+var reHomeWorkspaceChar = regexp.MustCompile("[^a-z0-9-]")
+
+// getHomeWorkspaceName returns the full name of the home workspace for a given user
+// This full name includes home bucket workspaces whose name is based on the user name
+// sha1 hash.
+func (h *homeWorkspaceHandler) getHomeWorkspaceName(userName string) logicalcluster.Name {
+	// bucketLevels <= 5
+	// bucketSize <= 4
```

Comment:
what does this mean?

Reply:
if these are required for correct function, you must check this and return an error

Reply:
My plan was to test that in the related CLI options.
And I didn't want the signature to return 2 values...

An intermediate option would be to set the values to the Max value if it is upper then the max value.
And complement that with a decent check and error message in the KCP CLI options.
WDYM ?